File: problems/217-contains-duplicate/contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const mapper = new Map()

    for(let i = 0; i < nums.length; i++) {
        if(mapper.has(nums[i])) {
            return true
        }
        
        mapper.set(nums[i], 1)
    }

    return false
};

File: problems/217-contains-duplicate/contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """

        freq = {}

        for num in nums:
            if freq.get(num):
                return True

            freq[num] = True

        return False
        